src/hooks/useArbTokenBridge.ts

```
@@ -421,10 +461,71 @@ export const useArbTokenBridge = (
       }
 
       const receipt = await tx.wait()
+      updateTokenBalances()
+
+      const { hash } = tx
+      if (!hash) return receipt
```

I think this control flow could be clearer by only conditionally executing the `getMessageResult` promise. the receipt gets returned either way, the hash just determins if additional work is done

---

isn't that already the case? i.e., `getMessageResult` only gets executed conditionally?

---

yes, sorry I didn’t explain this correctly. what I’m trying to say - the early return with the receipt is unnecessarily confusing IMO, it does the same thing as the return at the end of the function but is presented as separate logic. I think it would be better expressed by wrapping the getMessageResult in an if statement, e.g (if hash exists getMessageResult else throw error). or if the hash not being present should stop the function from continuing, it should throw an error

---

ah